mne/decoding/transformer.py

```diff
@@ -102,19 +125,31 @@ def transform(self, epochs_data, y=None):
         X : array, shape (n_epochs, n_channels, n_times)
             The data concatenated over channels.
         """
+        from mne.cov import _apply_scaling_array
         if not isinstance(epochs_data, np.ndarray):
             raise ValueError("epochs_data should be of type ndarray (got %s)."
                              % type(epochs_data))
 
         X = np.atleast_3d(epochs_data)
+        n_epochs, n_channels, n_times = X.shape
+        X = X.transpose(1, 0, 2)
 
-        for key, this_pick in six.iteritems(self.picks_list_):
-            if self.with_mean:
-                X[:, this_pick, :] -= self.ch_mean_[key]
-            if self.with_std:
-                X[:, this_pick, :] /= self.std_[key]
+        if self.scaler == 'standard':
+            for idx, (_, pick) in enumerate(self._pick_types):
+                if self.with_mean:
+                    X[pick] -= self.ch_mean_[idx]
+                if self.with_std:
+                    X[pick] /= self.std_[idx]
 
-        return X
+        elif self.scaler == 'default':
+            _apply_scaling_array(X, self._pick_types, dict())
+
+        else:
+            X = X.reshape(n_channels, -1)
```

Comment:
add explicit check isattr(self, self.scalers_ ), else throw error self 'has not been fitted'